pkg/daemon/cleanup/cleanup.go

```diff
@@ -107,6 +120,35 @@ func removeServices(svcMgr manager.Manager, services []servicescm.Service) error
 	return nil
 }
 
+// waitForProcessToStop waits until no processes with the given name exist
+func waitForProcessToStop(name string) error {
+	return wait.PollImmediate(retry.WindowsAPIInterval, retry.ResourceChangeTimeout, func() (done bool, err error) {
+		return !processRunning(name), nil
+	})
+}
+
+// processName returns the name of a process, given a service's command
+func processName(command string) (string, error) {
+	// The process name is the file name, without the .exe extension.
+	exeSplit := strings.Split(command, ".exe")
+	if len(exeSplit) < 2 {
+		return "", fmt.Errorf("command does not include '.exe'")
+	}
+	return filepath.Base(exeSplit[0]), nil
+}
+
+// processRunning returns true if the given process is running
+func processRunning(name string) bool {
+	// Assume that an error means the process doesn't exist.
+	// A better solution for this would be to call CreateToolhelp32Snapshot and iterate through the processes
```

Comment:
The better solution seems like a heavy hammer. What is the advantage of doing that? Can't you get the [StatusHandle()](https://cs.opensource.google/go/x/sys/+/v0.4.0:windows/svc/service.go;l=297) of the service which contains the pid and the use that to wait for the process to stop?